File: RepoTransBench/target_projects/C/Python/tinycthread_tinycthread/public_tests/test_tinycthread_api_public.py
import threading
import time

def api_public_thread_func(sum_arg):
    time.sleep(0.009) # 9ms
    sum_arg[0] += 5
    return 42

def test_api_public_thread_func():
    sum_val = [33]
    ret_val = [None]
    def thread_wrapper():
        ret_val[0] = api_public_thread_func(sum_val)
    t = threading.Thread(target=thread_wrapper)
    t.start()
    t.join()
    assert sum_val[0] == 38
    assert ret_val[0] == 42

def test_thrd_yield():
    # Should always succeed; in Python, time.sleep(0) yields
    time.sleep(0)
    assert True

def test_thrd_current():
    self = threading.current_thread()
    assert self is not None

File: RepoTransBench/target_projects/C/Rust/tinycthread_tinycthread/public_tests/test_tinycthread_api_public.rs
use std::sync::{Arc, Mutex};
use std::thread;
use std::time::Duration;

fn api_public_thread_func(sum: Arc<Mutex<i32>>) -> i32 {
    thread::sleep(Duration::from_millis(9));
    let mut sum = sum.lock().unwrap();
    *sum += 5;
    42
}

#[test]
fn test_api_public_thread_func() {
    let res = Arc::new(Mutex::new(33));
    let res_clone = res.clone();
    let handle = thread::spawn(move || api_public_thread_func(res_clone));
    let retval = handle.join().unwrap();
    assert_eq!(*res.lock().unwrap(), 38);
    assert_eq!(retval, 42);

    thread::yield_now();

    let id = thread::current().id();
    assert!(format!("{:?}", id).len() > 0);
}